armi/reactor/converters/uniformMesh.py

```diff
@@ -1130,6 +1160,89 @@ def _calculateReactionRates(lib, keff, assem):
                 continue
             globalFluxInterface.calcReactionRates(b, keff, lib)
 
+    def _calcReactionRatesBlockList(objList, keff, xsNucDict):
+        r"""
+        Compute 1-group reaction rates for the objects in objList (usually a block).
+
+        Parameters
+        ----------
+        objList : List[Block]
+            The list of objects to compute reaction rates on. Notionally this could be upgraded to be
+            any kind of ArmiObject but with params defined as they are it currently is only
+            implemented for a block.
+
+        keff : float
+            The keff of the core. This is required to get the neutron production rate correct
+            via the neutron balance statement (since nuSigF has a 1/keff term).
+
+        xsNucDict: Dict[str, XSNuclide]
+            Microscopic cross sections to use in computing the reaction rates. Keys are
+            nuclide names (e.g., "U235") and values are the associated XSNuclide objects
+            from the cross section library, which contain the microscopic cross section
+            data for a given nuclide in the current cross section group.
+
+        .. impl:: Return the reaction rates for a given ArmiObject
+            :id: I_ARMI_FLUX_RX_RATES_BY_XS_ID
+            :implements: R_ARMI_FLUX_RX_RATES
+
+            This is an alternative implementation of :need:`I_ARMI_FLUX_RX_RATES` that
+            is more efficient when computing reaction rates for a large set of blocks
+            that share a common set of microscopic cross sections.
+
+            For more detail on the reation rate calculations, see :need:`I_ARMI_FLUX_RX_RATES`.
+        """
+        from armi.physics.neutronics.globalFlux.globalFluxInterface import (
+            RX_PARAM_NAMES,
+            RX_ABS_MICRO_LABELS,
+        )
```

Comment:
Yeah, the import isn't great here but I can't import at the top of the file because of a circular import.

We could explicitly redefine `RX_PARAM_NAMES` and `RX_ABS_MICRO_LABELS` here, but then there's a risk that those strings get changed over in `globalFluxInterface` but not here.